plugin.py

```diff
@@ -132,6 +147,141 @@ def run(self, edit: sublime.Edit) -> None:
         session.send_request(Request("rust-analyzer/reloadWorkspace"))
 
 
+class RustAnalyzerMemoryUsage(LspTextCommand):
+    session_name = "rust-analyzer"
+
+    def run(self, edit: sublime.Edit) -> None:
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+
+        session.send_request(Request("rust-analyzer/memoryUsage"), self.on_result)
+
+    def on_result(self, payload: str) -> None:
+        window = self.view.window()
+        if window is None:
+            return
+        sheets = window.selected_sheets()
+        view = window.new_file(flags=sublime.TRANSIENT)
+        view.set_scratch(True)
+        view.set_name("--- RustAnalyzer Memory Usage ---")
+        view.run_command("append", {"characters": "Per-query memory usage:"})
+        for line in payload.splitlines():
+            view.run_command("append", {"characters": '\n{}'.format(line)})
+        view.run_command("append", {"characters": "\n(note: database has been cleared)"})
+        view.set_read_only(True)
+        sheet = view.sheet()
+        if sheet is not None:
+            sheets.append(sheet)
+            window.select_sheets(sheets)
+
+
+Runnable = TypedDict('Runnable', {
+    'label': str,
+    'args': {
+        'executableArgs': List[str],
+        'workspaceRoot': str,
+        'overrideCargo': Optional[str],
+        'cargoArgs': List[str],
+        'cargoExtraArgs': List[str]},
+    'kind': str,
+}
+)
+
+
+class RustAnalyzerExec(LspTextCommand):
+    session_name = "rust-analyzer"
+
+    def run(self, edit: sublime.Edit) -> None:
+        params = text_document_position_params(self.view, self.view.sel()[0].b)
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+        session.send_request(Request("experimental/runnables", params), self.on_result)
+
+    def run_termius(self, check_phrase: str, payload: List[Runnable]) -> None:
+        window = self.view.window()
+        if window is None:
+            return
+
+        if len(payload) == 0:
+            return
+
+        output = None
+        for i in range(len(payload)):
+            if payload[i]["label"].startswith(check_phrase):
+                output = payload[i]
+
+        if not output:
+            return
+
+        if not Terminus:
+            sublime.error_message(
+                'Cannot run executable "{}": You need to install the "Terminus" package and then restart Sublime Text'.format(output["kind"]))
+            return
+        if output["args"]["overrideCargo"]:
+            cargo_path = output["args"]["overrideCargo"]
+        else:
+            cargo_path = '"{}"'.format(shutil.which("cargo"))
```

Comment:
Should we also return with an error message if `cargo` is missing?

Reply:
Yeah that seems logical